src/inferences/junctionTree/create-initial-potentials.ts

```diff
@@ -100,9 +124,12 @@ const createCliquePotentials = (clique: IClique, network: INetwork, given: IComb
 export default (cliques: IClique[], network: INetwork, given: ICombinations): ICliquePotentials => {
   const cliqueFactors = createICliqueFactors(cliques, network)
 
-  return reduce(
-    (acc, clique) => assoc(clique.id, createCliquePotentials(clique, network, given, cliqueFactors), acc),
-    {},
-    cliques,
-  )
+  const cliquePotentials: ICliquePotentials = {}
```

Comment:
The use of assoc inside the reduce results in unnecessary copying of the accumulator in each iteration of the reduce.    Replacement of this with mutation in place offers small performance improvmeents.

Reply:
I'm always avoiding mutations but since it happens internally (only inside this function) it's okay.